src/test/java/io/jenkins/security/CompatibilityTest.java

```diff
@@ -0,0 +1,109 @@
+package io.jenkins.security;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.stream.Stream;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+import org.apache.commons.io.IOUtils;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.fail;
+
+public class CompatibilityTest {
+    private static Stream<Arguments> test() {
+        return Stream.of(
+                "4.0",
+                "4.0.1",
+                "4.1",
+                "4.2",
```

Comment:
You could just scan `test/classes/old-remoting/*.jar` rather than hard-coding all these.